core/src/main/scala/kafka/coordinator/transaction/TransactionMarkerChannel.scala

```diff
@@ -41,7 +40,41 @@ class TransactionMarkerChannel(interBrokerListenerName: ListenerName,
   // we need the metadataPartition so we can clean up when Transaction Log partitions emigrate
   case class PendingTxnKey(metadataPartition: Int, producerId: Long)
 
-  private val brokerStateMap: concurrent.Map[Int, DestinationBrokerAndQueuedMarkers] = concurrent.TrieMap.empty[Int, DestinationBrokerAndQueuedMarkers]
+  class BrokerRequestQueue(private var destination: Node) {
+
+    // keep track of the requests per metadata partition so we can easily clear the queue
+    // during partition emigration
+    private val requestsPerMetadataPartition: concurrent.Map[Int, BlockingQueue[TxnMarkerEntry]]
```

Comment:
@guozhangwang need to do this per metadataPartition so we can drain the queue on partition emigration